### DISCUSSION TITLE
Getting the root path of the app in sails controllers

### DISCUSSION THREAD

**Discussion by MaheshkumarSundaram** [neonexus/sails-react-bootstrap-webpack#82] [General]
Hi @neonexus,

Say my root folder is: "F:/myProjects/myWebApp/".

I would like to get this root path in any of Sails controllers.

I know Node.js has a lot of options. But it seems most of them may or may not work based on how I run the app and other things.

Could you please let me know if there is any reliable way to get the root path?

Thanks!

**Comment by neonexus** [comment]
Here's a quick n dirty helper (put in `api/helpers`):

```javascript
const { resolve } = require('path');

module.exports = {
    sync: true, // this is a synchronous helper

    friendlyName: 'Get Project Root Directory',

    description: 'Returns the project root directory',

    inputs: {},

    exits: {},

    fn: function(inputs, exits) {
        // __dirname here will be something like `/home/user/my-project/api/helpers`
        // path.resolve() here will return something like `/home/user/my-project`
        return exits.success(resolve(__dirname, '../../'));
    }
};
```

**Reply by MaheshkumarSundaram** [reply]
Eventually I would do the same. Just thought to ask you if there is any proper way.
But thank you anyways.
